### PR TITLE
Remove disable efb copy

### PR DESCRIPTION
**Pull request by degasus**

### CODE REVIEW

File: Source/Core/DolphinWX/VideoConfigDiag.cpp [hunk 1]
```diff
@@ -123,9 +123,7 @@ static wxString borderless_fullscreen_desc = wxTRANSLATE("Implement fullscreen m
...
-static wxString efb_copy_desc = wxTRANSLATE("Disable emulation of EFB copies.\nThese are often used for post-processing or render-to-texture effects, so while checking this setting may give a minor speedup over EFB to Texture it almost invariably also causes issues.\n\nIf unsure, leave this unchecked.");
-static wxString efb_copy_texture_desc = wxTRANSLATE("Store EFB copies in GPU texture objects.\nThis isn't particularly accurate, but it works well enough for most games and gives a great speedup over EFB to RAM.\n\nIf unsure, leave this checked.");
-static wxString efb_copy_ram_desc = wxTRANSLATE("Accurately emulate EFB copies.\nNumerous games depend on this for certain graphical effects or gameplay functionality.\nThis is much slower than EFB to Texture.\n\nIf unsure, check EFB to Texture instead.");
+static wxString skip_efb_copy_to_ram_desc = wxTRANSLATE("Skip GPU syncing on EFB copys. Causes graphical defects in a small number of other games.\n\nIf unsure, leave this checked.");
```
Comment (degasus) [inline Source/Core/DolphinWX/VideoConfigDiag.cpp]: Whoever wants to write this text, feel free to send me a better one.

File: Source/Core/VideoCommon/VideoConfig.cpp [hunk 2]
```diff
@@ -97,8 +97,7 @@ void VideoConfig::Load(const std::string& ini_file)
 
 	IniFile::Section* hacks = iniFile.GetOrCreateSection("Hacks");
 	hacks->Get("EFBAccessEnable", &bEFBAccessEnable, true);
-	hacks->Get("EFBCopyEnable", &bEFBCopyEnable, true);
-	hacks->Get("EFBToTextureEnable", &bCopyEFBToTexture, true);
+	hacks->Get("EFBToTextureEnable", &bSkipEfbCopy2Ram, true);
```
Comment (degasus) [inline Source/Core/VideoCommon/VideoConfig.cpp]: I'll rename this ini option when the current open ini PR is merged. Too many conflicts right now.

**Comment by neobrain** [comment]
<3

**Comment by delroth** [comment]
Can we have JMC and MaJoR report if there are still some games that require this to be playable?

**Comment by degasus** [comment]
@delroth JMC47 did ask for removing this option: https://code.google.com/p/dolphin-emu/issues/detail?id=8254

**Comment by delroth** [comment]
Oh.

File: Source/Core/DolphinWX/VideoConfigDiag.cpp (hunk 1 above)
Comment (delroth) [inline Source/Core/DolphinWX/VideoConfigDiag.cpp]: synchronizing*
copies*
Remove "other"

**Comment by MaJoRoesch** [comment]
The last thing that disabling efb copies helped (that I know of) was [issue 5920](https://code.google.com/p/dolphin-emu/issues/detail?id=5920), and that was fixed by TFN. I can't think of any others, and some searching on the wiki didn't find any either.

So I say, let's do this!

**Comment by Linktothepast** [comment]
Agreed, there is nothing that disabling efb copies really fix. There was also the case of resident evil code veronica in the past but that was properly fixed too.

**Comment by Sonicadvance1** [comment]
woo

**Comment by JMC47** [comment]
Resident Evil Code Veronica works with EFB2RAM and EFB2Tex.  Feel free to merge.

File: Source/Core/Core/Movie.cpp [hunk 3]
```diff
@@ -1211,8 +1211,8 @@ void SaveRecording(const std::string& filename)
 	strncpy((char *)header.videoBackend, s_videoBackend.c_str(),ArraySize(header.videoBackend));
 	header.CPUCore = s_iCPUCore;
 	header.bEFBAccessEnable = g_ActiveConfig.bEFBAccessEnable;
-	header.bEFBCopyEnable = g_ActiveConfig.bEFBCopyEnable;
-	header.bCopyEFBToTexture = g_ActiveConfig.bCopyEFBToTexture;
+	header.bEFBCopyEnable = true;
+	header.bSkipEfbCopyToRam = g_ActiveConfig.bSkipEfbCopyToRam;
```
Comment (delroth) [inline Source/Core/Core/Movie.cpp]: s/Efb/EFB/. We are using EFB for all other fields of that structure.

File: Source/Core/DolphinWX/VideoConfigDiag.cpp [hunk 4]
```diff
@@ -479,21 +477,10 @@ VideoConfigDiag::VideoConfigDiag(wxWindow* parent, const std::string &title, con
...
-
-	group_efbcopy->Add(efbcopy_disable, 0, wxLEFT | wxRIGHT | wxBOTTOM, 5);
-	group_efbcopy->AddStretchSpacer(1);
-	group_efbcopy->Add(efbcopy_texture, 0, wxRIGHT, 5);
-	group_efbcopy->Add(efbcopy_ram, 0, wxRIGHT, 5);
-
 	szr_efb->Add(CreateCheckBox(page_hacks, _("Skip EFB Access from CPU"), wxGetTranslation(efb_access_desc), vconfig.bEFBAccessEnable, true), 0, wxBOTTOM | wxLEFT, 5);
 	szr_efb->Add(CreateCheckBox(page_hacks, _("Ignore Format Changes"), wxGetTranslation(efb_emulate_format_changes_desc), vconfig.bEFBEmulateFormatChanges, true), 0, wxBOTTOM | wxLEFT, 5);
-	szr_efb->Add(group_efbcopy, 0, wxEXPAND | wxALL, 5);
+	szr_efb->Add(CreateCheckBox(page_hacks, _("Skip EFB copys to RAM"), wxGetTranslation(skip_efb_copy_to_ram_desc), vconfig.bSkipEfbCopyToRam), 0, wxBOTTOM | wxLEFT, 5);
```
Comment (delroth) [inline Source/Core/DolphinWX/VideoConfigDiag.cpp]: copies

**Comment by Sonicadvance1** [comment]
http://hastebin.com/itipekopoc.drps
If you could also remove it from the Android UI using the patch in the hastebin.

**Comment by delroth** [comment]
@Sonicadvance1 given that the Android UI still mentions the cache option, I would rather this be all cleaned up in a separate PR.

**Comment by Sonicadvance1** [comment]
Sounds fine.

**Pull request merged**